server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -85,6 +85,26 @@
     public SearchResponse() {
     }
 
+    public SearchResponse(StreamInput in) throws IOException {
```

Comment:
Can we also remove an empty constructor and make all class variables final here?

Reply:
yes this is not so hard so I am making this change.